plugins/modules/dcnm_service_node.py

```diff
@@ -0,0 +1,745 @@
+    - Service Nodes that are not specified in the playbook will be untouched.
+
+Replaced:
+  Service Nodes defined in the playbook will be replaced in the service fabric.
+    - If the service node does not exist it will not be added.
+    - If the service node exists but properties managed by the playbook are different
+      they will be updated if possible.
+    - Properties that can be managed by the module but are not specified
+      in the playbook will be deleted or defaulted if possible.
+    - Service node that are not specified in the playbook will be untouched.
+
+Overridden:
+  Service Node defined in the playbook will be overridden in the service fabric.
+    - If the service node does not exist it will be added.
+    - If the service node exists but properties managed by the playbook are different
+      they will be updated if possible.
+    - Properties that can be managed by the module but are not specified
+      in the playbook will be deleted or defaulted if possible.
+    - Service Node that are not specified in the playbook will be deleted.
+
+Deleted:
+  Service Node defined in the playbook will be deleted.
+  If no Service Node are provided in the playbook, all service node present on that DCNM fabric will be deleted.
+
+Query:
+  Returns the current DCNM state for the service node listed in the playbook.
+
+- name: Merge Service Nodes
+  cisco.dcnm.dcnm_service_node:
+    fabric: Fabric1
+    service_fabric: external
+    state: merged 
+    config:
+    - name: SN-11
+      type: firewall
+      form_factor: virtual
+      svc_int_name: svc1
+      attach_interface: Ethernet1/1
+      switches:
+      - 192.168.1.224
+    - name: SN-12
+      type: firewall
+      form_factor: virtual
+      svc_int_name: svc1
+      attach_interface: vPC1
+      switches:  # up to two switches, if two switches are provided, vpc is only option
+      - 192.168.1.224
+      - 192.168.1.225
+
+- name: Replace Service Nodes form factor/type parameter
+  cisco.dcnm.dcnm_service_node:
+    fabric: Fabric1
+    service_fabric: external
+    state: replaced
+    config:
+    - name: SN-11
+      type: firewall
+   #  Replace can only modify the form factor
+   #  form_factor: virtual  # the virtual will be changed to new physical
+   #  form_factor: physical
+      svc_int_name: svc1
+      attach_interface: Ethernet1/1
+      switches:
+      - 192.168.1.224
+   #   Nothing will be replaced in the below service node as there is no change 
+   #   Dont touch this if its present on DCNM
+   # - name: SN-12
+   #   type: firewall
+   #   form_factor: virtual
+   #   svc_int_name: svc1
+   #   attach_interface: vPC1
+   #   switches:  # up to two switches, if two switches are provided, vpc is only option
+   #   - 192.168.1.224
+   #   - 192.168.1.225
+
+- name: Override Service Nodes
+  cisco.dcnm.dcnm_service_node:
+    fabric: Fabric1
+    service_fabric: external
+    state: overridden
+    config:
+   # Create this service node
```

Comment:
Is there a case where we will not give any config, but override

Reply:
no config is mandatory for override.